app/src/main/java/zapsolutions/zap/fragments/AdvancedSettingsFragment.java

```diff
@@ -45,6 +47,17 @@ public boolean onPreferenceChange(Preference preference, Object newValue) {
             }
         });
 
+        // Request exchange rates when the provider changed
+        ListPreference listExchangeRateProvider = findPreference("exchangeRateProvider");
+        listExchangeRateProvider.setOnPreferenceChangeListener(new Preference.OnPreferenceChangeListener() {
+            @Override
+            public boolean onPreferenceChange(Preference preference, Object newValue) {
+                PrefsUtil.edit().putString(PrefsUtil.EXCHANGE_RATE_PROVIDER, newValue.toString()).commit();
```

Comment:
If `setOnPreferenceChangeListener` is invoked, doesn't it imply that the value already has been changed and this line changes it again?

Reply:
Yes, but I want to update the exchange rates when changing the provider. 
If I don't have that line here, it will update the exchange rates before the provider is switched and therefore fetching the rates from the wrong provider.

Reply:
Or in other words: The value is not changed until true is returned. That's why I have to do it earlier.